modules/shared/src/main/scala/org/tessellation/schema/transaction.scala

```diff
@@ -41,7 +41,14 @@ object transaction {
   @newtype
   case class TransactionSalt(value: Long)
 
-  // TODO: figure out the Fiber usage for transaction
+  @derive(decoder, encoder, eqv, show)
+  case class TransactionData(
+    source: Address,
+    destination: Address,
+    amount: TransactionAmount,
+    fee: TransactionFee
+  )
```

Comment:
Depending on what a `Fiber` should be, possibly `Salt` and `TransactionReference` should also be part of the data. They influence the transaction hash.

Reply:
Though I don't have the full understanding of `Fiber` so just posting to get it discussed.

Reply:
For example: Transaction would keep the TransactionReference as its first polymorphic type